src/lib/FederatorERC.ts

```diff
@@ -390,71 +397,119 @@ export default class FederatorERC extends Federator {
+                err,
             );
-            if (fs.existsSync(revertedTxnsPath)) {
-                revertedTx = await AppDataSource.getRepository(FailedTransactions)
-                    .findOne({
-                        where: {
-                            mainChain: voteTransactionParams.mainChainId,
-                            sideChain: voteTransactionParams.sideChainId,
-                            transactionId: voteTransactionParams.transactionId
-                        }
-                    });
-            }
+        }
+    }
 
-            if (revertedTx) {
-                this.logger.warn(
-                    `Skipping Voting ${voteTransactionParams.amount} of originalTokenAddress:${voteTransactionParams.tokenAddress}
-          TransactionId ${voteTransactionParams.transactionId} since it's marked as reverted.`,
-                    revertedTx.txData,
-                );
-                return false;
+    async verifyIfwasRevertedAndRetry(params, txAbi) {
+        const revertedTx = await AppDataSource.getRepository(FailedTransactions)
+          .findOne({
+              where: {
+                  mainChain: params.mainChainId,
+                  sideChain: params.sideChainId,
+                  transactionId: params.transactionId
+              }
+          });
+
+        const result = await this.validateAndVote(params, txAbi);
+
+        if (revertedTx && (result.voteSuccess || result.wasVotedBefore || result.wasProcessed)) {
+            await AppDataSource.getRepository(FailedTransactions).delete({
+                    mainChain: params.mainChainId,
+                    sideChain: params.sideChainId,
+                    transactionId: params.transactionId
+            });
+        }
+    }
+
+    async validateAndVote(params, txDataAbi): Promise<ValidateAndVoteReturn> {
+        const validateAndVoteReturn: ValidateAndVoteReturn = {} as ValidateAndVoteReturn;
+
+        const fedAddress = await params.transactionSender.getAddress(this.config.privateKey);
+
+        const hasVoted = await params.sideFedContract
+          .hasVoted(params.transactionId, fedAddress);
+
+        const hasVotedDb = await AppDataSource.getRepository(Votes).findOne({
+            where: {transactionId: params.transactionId}});
+
+        const wasProcessed = await params.sideFedContract
+          .transactionWasProcessed(params.transactionId);
+
+        if(hasVoted || wasProcessed || hasVotedDb) {
+            this.logger.warn(`Transaction ${params.transactionId} will not be voted by the 
+                federator: ${fedAddress} hasVoted result ${hasVoted} - wasProcessed result ${wasProcessed} hasVotedDb 
+                result ${hasVotedDb}`);
+            if(hasVoted || hasVotedDb) {
+                validateAndVoteReturn.wasVotedBefore = true;
+            }
+            if(wasProcessed) {
+                validateAndVoteReturn.wasProcessed = true;
```

Comment:
can be converted to `validateAndVoteReturn.wasProcessed = wasProcessed` to clean the nested if, the same for the one above this one